web/api/v1/api_test.go

```diff
@@ -613,6 +613,91 @@ func testEndpoints(t *testing.T, api *API, tr *testTargetRetriever, testLabelAPI
 				},
 			},
 		},
+		{
+			endpoint: api.query,
+			query: url.Values{
+				"query": []string{"prometheus_target_interval_length_seconds"},
+			},
+			response: &queryData{
+				ResultType: parser.ValueTypeVector,
+				Result: promql.Vector{
+
+					{
+						Point: promql.Point{
+							V: 2.000542683,
+							T: timestamp.FromTime(api.now()),
+						},
+						Metric: nil,
+					},
+					{
+						Point: promql.Point{
+							V: 2.000355733,
+							T: timestamp.FromTime(api.now()),
+						},
+						Metric: nil,
+					},
+					{
+						Point: promql.Point{
+							V: 2.00001554,
+							T: timestamp.FromTime(api.now()),
+						},
+						Metric: nil,
+					},
+					{
+						Point: promql.Point{
+							V: 1.999594005,
+							T: timestamp.FromTime(api.now()),
+						},
+						Metric: nil,
+					},
+					{
+						Point: promql.Point{
+							V: 1.9994983290000001,
+							T: timestamp.FromTime(api.now()),
+						},
+						Metric: nil,
+					},
+				},
+			},
+		},
+		{
+			endpoint: api.queryRange,
+			query: url.Values{
+				"query": []string{"prometheus_target_interval_length_seconds"},
+				"start": []string{"0"},
```

Comment:
How we are sure that the mocked data is not giving us series in the right order before sorting? The quick way is to remove sort and check out the result (: